scalafix-tests/unit/src/test/scala/scalafix/cli/CliSemanticTest.scala

```diff
@@ -0,0 +1,74 @@
+package scalafix.cli
+
+import scala.collection.immutable.Seq
+import scalafix.internal.cli.CommonOptions
+import scalafix.tests.BuildInfo
+
+class CliSemanticTest extends BaseCliTest {
+
+  checkSemantic(
+    name = "--classpath inference error",
+    args = Nil,
+    expectedExit = ExitStatus.InvalidCommandLineOption,
+    outputAssert = output => {
+      assert(output.contains("Unable to infer --classpath"))
+    }
+  )
+
+  checkSemantic(
+    name = "--classpath explicit is OK",
+    args = Seq(
+      "--classpath",
+      BuildInfo.semanticClasspath.getAbsolutePath
+    ),
+    expectedExit = ExitStatus.Ok
+  )
+
+  checkSemantic(
+    name = "--sourceroot is not a file",
+    args = Seq(
+      "--sourceroot",
+      "bogus",
+      "--classpath",
+      BuildInfo.semanticClasspath.getAbsolutePath
+    ),
+    expectedExit = ExitStatus.InvalidCommandLineOption,
+    outputAssert = msg => assert(msg.contains("Invalid --sourceroot"))
+  )
+
+  checkSemantic(
+    name = "--sourceroot points to non-existing file",
+    args = Seq(
+      "--sourceroot",
+      CommonOptions.default.workingDirectory,
+      "--classpath",
+      BuildInfo.semanticClasspath.getAbsolutePath
+    ),
+    expectedExit = ExitStatus.InvalidCommandLineOption,
+    outputAssert = msg => {
+      assert(msg.contains("is not a file"))
```

Comment:
cc/ @usrinivasan scalafix-cli will now fail fast when the --sourceroot maps semanticdb relative paths to non-existing files